From 268dd1c294e1bf41cffe04d5a43ef0fc40b875a5 Mon Sep 17 00:00:00 2001
From: Karthikeyan K <karthikk@habana.ai>
Date: Wed, 10 Nov 2021 04:33:13 +0200
Subject: [PATCH] [SW-36216] gaudi2: preboot: configure hbw rr

Access to hbw regions are blocked in preboot as per the hbw
specification. Only noc/psoc engine's rrs are blocked in preboot.
pcie rr was already blocked in ppboot.

Change-Id: Iad90533dfd95ac6bceca91a663e9237583870cd5
---

diff --git a/arch/gaudi2/common/security.c b/arch/gaudi2/common/security.c
index 6e48247..6dfa14f 100644
--- a/arch/gaudi2/common/security.c
+++ b/arch/gaudi2/common/security.c
@@ -601,15 +601,9 @@
 #define SRAM_RSVD_BASE	0x1000008000000000ull
 #define DRAM_RSVD_END	0xFF00000000000000ull
 
-static void ppboot_secure_hbw(void)
+static void hl_secure_hbw(struct hl_rr_eng_select rr_sel)
 {
 	u64 min, max;
-	struct hl_rr_eng_select rr_sel;
-
-	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
-
-	/* Only block pcie RR in ppboot */
-	rr_sel.pcie = true;
 
 	/*
 	 * SPI FLASH END to DBG SPACE BASE
@@ -652,6 +646,31 @@
 	hl_secure_hbw_rr(HL_SEC_HBW_HBM, min, max, rr_sel);
 }
 
+static void ppboot_secure_hbw(void)
+{
+	struct hl_rr_eng_select rr_sel;
+
+	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
+
+	/* Only block pcie RR in ppboot */
+	rr_sel.pcie = true;
+
+	hl_secure_hbw(rr_sel);
+}
+
+static void preboot_secure_hbw(void)
+{
+	struct hl_rr_eng_select rr_sel;
+
+	memset(&rr_sel, 0, sizeof(struct hl_rr_eng_select));
+
+	/* Block noc and psoc RR in preboot */
+	rr_sel.noc = true;
+	rr_sel.psoc = true;
+
+	hl_secure_hbw(rr_sel);
+}
+
 void ppboot_security_init(void)
 {
 	/* Initialize only if security is enabled */
@@ -672,4 +691,5 @@
 
 	hl_dev_prot_bits_init();
 	preboot_secure_lbw();
+	preboot_secure_hbw();
 }
